utils/http-utils/src/main/scala/pl/touk/nussknacker/http/backend/HttpClientConfig.scala

```diff
@@ -75,18 +82,49 @@ object HttpClientConfig {
       useNative = forOption[Boolean]("useNative"),
       followRedirect = forOption[Boolean]("followRedirect"),
       forceShutdown = forOption[Boolean]("forceShutdown"),
-      configForProcess = forOption("configForProcess")(mapValueReader(vr))
+      configForProcess = forOption("configForProcess")(mapValueReader(vr)),
+      forbiddenHosts = forOption[List[String]]("forbiddenHosts"),
     )
   })
 
 }
 
 object DefaultHttpClientConfig {
 
-  def apply(): HttpClientConfig = HttpClientConfig(None, None, None, None, None, None, None)
+  def apply(): HttpClientConfig = HttpClientConfig(None, None, None, None, None, None, None, None)
 
   val maxPoolSize: Int = 20
 
   val timeout: FiniteDuration = 10 seconds
 
+  implicit class ClientConfigBuilderExtension(val builder: DefaultAsyncHttpClientConfig.Builder) extends AnyVal {
+
+    def setForbiddenHostRequestFilter(forbiddenHosts: List[String]): DefaultAsyncHttpClientConfig.Builder =
+      if (forbiddenHosts.nonEmpty) {
+        addForbiddenHostsFilter(forbiddenHosts)
+      } else {
+        builder
+      }
+
+    private def addForbiddenHostsFilter(forbiddenHosts: List[String]): DefaultAsyncHttpClientConfig.Builder = {
+      val lowerCaseHosts = forbiddenHosts.map(_.toLowerCase)
+      val lowCaseHostsWithMaybeLoopback = if (lowerCaseHosts.contains("localhost")) {
+        InetAddress.getLoopbackAddress.getHostAddress :: lowerCaseHosts
+      } else {
+        lowerCaseHosts
+      }
+      val hostsAsSet = lowCaseHostsWithMaybeLoopback.toSet
+      builder.addRequestFilter(new RequestFilter {
+        override def filter[T](ctx: FilterContext[T]): FilterContext[T] = {
+          val hostName = ctx.getRequest.getUri.getHost.toLowerCase
+          if (hostsAsSet.contains(hostName)) {
```

Comment:
I think it could also be useful to block K8S services with hosts ending with `svc.cluster.local`. That would require `forbiddenHosts` to become a list of patterns, or alternatively, we could add a second configuration option called `forbiddenHostPatterns`. WDYT?